tensorflow/python/distribute/input_lib.py

```diff
@@ -124,24 +126,31 @@ def get_distributed_datasets_from_function(dataset_fn,
         `worker_device_pairs`.
     strategy: a `tf.distribute.Strategy` object, used to run all-reduce to
         handle last partial batch.
+    replication_mode: Replication mode for the input function.
 
   Returns:
     A distributed dataset instance.
   """
   if tf2.enabled():
-    return DistributedDatasetsFromFunction(
-        dataset_fn,
-        input_workers,
-        input_contexts,
-        strategy)
+    if replication_mode == distribute_lib.InputReplicationMode.PER_WORKER:
+      return DistributedDatasetsFromFunction(
+          dataset_fn,
+          input_workers,
+          input_contexts,
+          strategy)
+    else:
+          return DistributedDatasetsFromFunctionForReplicas(
```

Comment:
what breaks if you use `DistributedDatasetsFromFunction(..)` here instead? I was hoping that since input_workers are set correctly, it might just work as-is? Is this something you've tried?

Reply:
The __iter__ checks are quite different. Do you want to have separate checks per case?

Reply:
which `iter` checks?

Reply:
ping

Reply:
_create_iterators_per_replica_with_input_context vs. _create_iterators_per_worker.